app/scripts/analysis/tools/powercalc/PowercalcMainCtrl.ctrl.coffee

```diff
@@ -0,0 +1,1529 @@
+      @populations = data.populations
+      lab = data.chosenlab
+      if (lab is "none") or (lab is null)
+        console.log "hit"
+        @agent = data.chosenCol
+      else
+        @agent = data.chosenVar
+      @drive_data()
+
+    @$scope.$on 'powercalc:TwoTGUI_alpha', (event, data)=>
+      @TwoTGUI_alpha = data.alpha_in
+      @TwoTGUI_update()
+
+    @$scope.$on 'powercalc:TwoTGUI_data', (event, data)=>
+      @populations = data.populations
+      if (data.chosenCol.length is 2)
+        @comp_agents = data.chosenCol
+      else
+        @comp_agents = data.chosenVar
+      @drive_data()
+
+    @$scope.$on 'powercalc:change_mode', (event, data)=>
+      @deployed=data.deploy
+      d3.select("#Two_TGUI_graph").select("svg").remove()
+      @TwoTGUI_update()
+      @OneTGUI_update()
+      #@render_mathjax()
+      if !@deployed
+        $("#psigma1i").text("1: ")
+        $("#psigma2i").text("2: ")
+        $("#pn1i").text("1: ")
+        $("#pn2i").text("2: ")
+        $("#pmean1i").text("1: ")
+        $("#pmean2i").text("2: ")
+        $("#OneTGUI_N_disp").text(": ")
+        $("#OneTGUI_mean_disp").text(": ")
+        $("#OneTGUI_sigma_disp").text(": ")
+
+
+    @$scope.$on 'powercalc:updateDataPoints', (event, data) =>
+      @data = data.dataPoints
+
+  render_mathjax: () ->
+    # window.MathJax.Ajax.config.root="../node_modules/mathjax"
+    window.MathJax.Hub.Config({
+      extensions: ["tex2jax.js"],
+      jax: ["input/TeX", "output/HTML-CSS"],
+      tex2jax: {
+        inlineMath: [ ['$','$'], ["\\(","\\)"] ],
+        displayMath: [ ['$$','$$'], ["\\[","\\]"] ],
+        processEscapes: true
+      },
+      "HTML-CSS": { availableFonts: ["TeX"] }
+    })
+
+    # @MathJax.Hub.Config({
+    #   extensions: ["tex2jax.js"],
+    #   jax: ["input/TeX", "output/HTML-CSS"],
+    #   tex2jax: {
+    #     inlineMath: [ ['$','$'], ["\\(","\\)"] ],
+    #     displayMath: [ ['$$','$$'], ["\\[","\\]"] ],
+    #     processEscapes: true
+    #   },
+    #   "HTML-CSS": { availableFonts: ["TeX"] }
+    # });
+    
+  drive_data: () ->
+    if (@selectedAlgorithm is "Two-sample t test (general case)")
+      # check population length
+      if (Object.keys(@populations).length isnt 2)
+        window.alert("main: population length")
+        return
+
+      @TwoTGUI_receive_data()
+      @TwoTGUI_graph()
+      return
+
+    else if (@selectedAlgorithm is "One-Sample (or Paired) t Test")
+      # check population length
+      if (Object.keys(@populations).length isnt 1)
+        window.alert("main: population length")
+        return
+        
+      @OneTGUI_receive_data()
+      @OneTGUI_graph()
+      return
+
+  update_algo: (evt) ->
+    console.log(@selectedAlgorithm)
+    @selectedAlgorithm = evt.currentTarget.value
+    @msgService.broadcast 'powercalc:updateAlgorithm_back',
+      @selectedAlgorithm
+
+  #cfap function only
+  cfap_clk: (evt) ->
+    obj = evt.currentTarget
```

Comment:
Angular two-way binding updates your scope variables for you, so you should use [`ng-model`](https://docs.angularjs.org/api/ng/directive/ngModel) here and similar situations instead of the `event.target`